synapse/handlers/user_directory.py

```diff
@@ -148,16 +159,30 @@ def _do_initial_spam(self):
         room_ids = yield self.store.get_all_rooms()
 
         logger.info("Doing initial update of user directory. %d rooms", len(room_ids))
-        num_processed_rooms = 1
+        num_processed_rooms = 0
 
         for room_id in room_ids:
-            logger.info("Handling room %d/%d", num_processed_rooms, len(room_ids))
+            logger.info("Handling room %d/%d", num_processed_rooms+1, len(room_ids))
             yield self._handle_initial_room(room_id)
             num_processed_rooms += 1
             yield sleep(self.INITIAL_SLEEP_MS / 1000.)
 
         logger.info("Processed all rooms.")
 
+        if self.include_pattern:
+            num_processed_users = 0
+            user_ids = yield self.store.get_all_local_users()
```

Comment:
Do we want to include appservice users here?

Reply:
I'm also worried that pulling out millions of users is going to be painful.

Reply:
And its going to take matrix.org more than a day on the default settings to get through all its users

Reply:
hm, point. so your suggestion is to batch? and yes, we probably should be including AS users.